packages/blade/src/components/Tabs/TabItem.tsx

```diff
@@ -0,0 +1,166 @@
+/* eslint-disable @typescript-eslint/prefer-nullish-coalescing */
+import styled from 'styled-components';
+import get from 'lodash/get';
+import React from 'react';
+import { CompositeItem } from '@floating-ui/react';
+import type { TabItemProps, TabsProps } from './types';
+import { useTabsContext } from './TabsContext';
+import { backgroundColor, paddingY, paddingX, textColor } from './tabTokens';
+import { iconSizeMap, useTabsItemPropRestriction } from './utils';
+import { Text } from '~components/Typography';
+import { castWebType, makeBorderSize, makeMotionTime, makeSpace } from '~utils';
+import useInteraction from '~utils/useInteraction';
+import { useIsMobile } from '~utils/useIsMobile';
+import { makeAccessible } from '~utils/makeAccessible';
+
+const StyledTabButton = styled.button<{
+  size: TabsProps['size'];
+  isFullWidthTabItem?: TabsProps['isFullWidthTabItem'];
+  variant: NonNullable<TabsProps['variant']>;
+  isVertical: boolean;
+  isSelected: boolean;
+}>(({ theme, isSelected, size, variant, isFullWidthTabItem, isVertical }) => {
+  const isMobile = useIsMobile();
+  const isFilled = variant === 'filled';
+  const device = isMobile ? 'mobile' : 'desktop';
+  const orientation = isVertical ? 'vertical' : 'horizontal';
+  const border = isVertical ? 'borderLeft' : 'borderBottom';
+  const selectedState = isSelected ? 'selected' : 'unselected';
+  const _variant = variant === 'borderless' ? 'bordered' : variant;
+  const background = backgroundColor[selectedState][_variant];
+
+  const getColor = (value: string): string => {
+    if (value === 'transparent') return 'transparent';
+    return get(theme, value);
+  };
+
+  return {
+    appearance: 'none',
+    textDecoration: 'none',
+    border: 'none',
+    outline: 'none',
+    cursor: 'pointer',
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: isVertical ? 'left' : 'center',
+    gap: makeSpace(theme.spacing[3]),
+    width: isFullWidthTabItem ? '100%' : undefined,
+    paddingTop: makeSpace(get(theme, paddingY[_variant][orientation][device][size!])),
+    paddingBottom: makeSpace(get(theme, paddingY[_variant][orientation][device][size!])),
+    paddingLeft: makeSpace(get(theme, paddingX[_variant][orientation][device][size!])),
+    paddingRight: makeSpace(get(theme, paddingX[_variant][orientation][device][size!])),
```

Comment:
should we use media queries instead? since its not too dificult here and don't think so our `isMobile` is server-render compatible so there might be flash of different padding in SSR.

Reply:
Okay, can you check this? 

https://github.com/razorpay/blade/pull/1694/commits/01ce3aab88dd7d69ebf5ce15ec65a79f87e54c05